java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1605,50 +1492,44 @@ public ExpressionTree binaryExpression1(ExpressionTree expression, Optional<List
     ExpressionTree result = expression;
 
     if (postfixOperatorAstNode.isPresent()) {
-      InternalSyntaxToken postfixOperatorToken = InternalSyntaxToken.create(postfixOperatorAstNode.get());
+      InternalSyntaxToken postfixOperatorToken = postfixOperatorAstNode.get();
       result = new InternalPostfixUnaryExpression(kindMaps.getPostfixOperator((JavaPunctuator) postfixOperatorAstNode.get().getType()), result, postfixOperatorToken);
     }
 
     return result;
   }
 
-  public ExpressionTree newTildaExpression(AstNode tildaTokenAstNode, ExpressionTree expression) {
-    InternalSyntaxToken operatorToken = InternalSyntaxToken.create(tildaTokenAstNode);
-    return new InternalPrefixUnaryExpression(Kind.BITWISE_COMPLEMENT, operatorToken, expression);
+  public ExpressionTree newTildaExpression(InternalSyntaxToken tildaToken, ExpressionTree expression) {
+    return new InternalPrefixUnaryExpression(Kind.BITWISE_COMPLEMENT, tildaToken, expression);
   }
 
-  public ExpressionTree newBangExpression(AstNode bangTokenAstNode, ExpressionTree expression) {
-    InternalSyntaxToken operatorToken = InternalSyntaxToken.create(bangTokenAstNode);
-    return new InternalPrefixUnaryExpression(Kind.LOGICAL_COMPLEMENT, operatorToken, expression);
+  public ExpressionTree newBangExpression(InternalSyntaxToken bangToken, ExpressionTree expression) {
+    return new InternalPrefixUnaryExpression(Kind.LOGICAL_COMPLEMENT, bangToken, expression);
   }
 
-  public ExpressionTree completeCastExpression(AstNode openParenTokenAstNode, TypeCastExpressionTreeImpl partial) {
-    return partial.complete(InternalSyntaxToken.create(openParenTokenAstNode));
+  public ExpressionTree completeCastExpression(InternalSyntaxToken openParenTokenAstNode, TypeCastExpressionTreeImpl partial) {
+    return partial.complete(openParenTokenAstNode);
   }
 
-  public TypeCastExpressionTreeImpl newBasicTypeCastExpression(PrimitiveTypeTreeImpl basicType, AstNode closeParenTokenAstNode, ExpressionTree expression) {
-    InternalSyntaxToken closeParenToken = InternalSyntaxToken.create(closeParenTokenAstNode);
+  public TypeCastExpressionTreeImpl newBasicTypeCastExpression(PrimitiveTypeTreeImpl basicType, InternalSyntaxToken closeParenToken, ExpressionTree expression) {
     return new TypeCastExpressionTreeImpl(basicType, closeParenToken, expression);
   }
 
-  public TypeCastExpressionTreeImpl newClassCastExpression(TypeTree type, Optional<List<Tuple<InternalSyntaxToken, Tree>>> classTypes, AstNode closeParenTokenAstNode,
-    ExpressionTree expression) {
-    InternalSyntaxToken closeParenToken = InternalSyntaxToken.create(closeParenTokenAstNode);
 
-    ImmutableList.Builder<Tree> boundsBuilder = ImmutableList.<Tree>builder();
+  public TypeCastExpressionTreeImpl newClassCastExpression(TypeTree type, Optional<List<Tuple<InternalSyntaxToken, Tree>>> classTypes, InternalSyntaxToken closeParenToken, ExpressionTree expression) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 200 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)